### PR TITLE
MRG: Improve memory footprint of _TimeViewer

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR improves the memory footprint of `_TimeViewer` by decreasing the reference count and solving reference cycles.

This is the comparison between `master` (black curve) and this PR (blue curve) using the following script with `mprof`:

```py
import os
import mne
import pyvista
from mne.datasets import sample
data_path = sample.data_path()
sample_dir = os.path.join(data_path, 'MEG', 'sample')
subjects_dir = os.path.join(data_path, 'subjects')
fname_stc = os.path.join(sample_dir, 'sample_audvis-meg')
stc = mne.read_source_estimate(fname_stc, subject='sample')
initial_time = 0.1
mne.viz.set_3d_backend('pyvista')
for j in range(5):
    for i in range(5):
        brain = stc.plot(subjects_dir=subjects_dir, initial_time=initial_time,
                         clim=dict(kind='value', pos_lims=[3, 6, 9]),
                         hemi='lh',
                         time_viewer=True, show_traces=True)
    pyvista.close_all()
```

image

There is probably room for improvement but this is a step forward.

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
I don't think I can go further with this method, there is the latest result:

image

File: mne/viz/_brain/_timeviewer.py [hunk 1]
```diff
@@ -837,6 +838,50 @@ def help(self):
...
+        if hasattr(self, "mpl_canvas"):
+            self.mpl_canvas.close()
+            self.mpl_canvas.time_viewer = None
+            self.mpl_canvas.canvas = None
+            self.mpl_canvas = None
+        self.time_actor = None
+        self.picked_renderer = None
+        self.act_data["lh"] = None
+        self.act_data["rh"] = None
+        self.act_data = None
```
Comment (larsoner) [inline mne/viz/_brain/_timeviewer.py]: Oh wow :)

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_timeviewer.py]: I took no risk xD

**Comment by larsoner** [comment]
Failure is just a timeout, thanks @GuillaumeFavelier

**Pull request merged**